### PR TITLE
daemon: fix endpoint restoration

### LINKED ISSUE #18923: Fix endpoint restoration with CEP CRD disabled
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

With CEP CRD disabled, the Cilium K8s pod watcher watches for all pods rather than just the local pods. However, if we take a look at [validateEndpoint()](https://github.com/cilium/cilium/blob/457684158d325b3e38e83b4d64717c34c4e2cad2/daemon/cmd/state.go#L48), then it's possible for Cilium to think that an endpoint that previously existed locally but moved to another node is a validate endpoint to restore _locally_.  This is wrong and this check should be expanded to check whether the pod returned from `GetCachedPod` resides on the local node first.

### Cilium Version

master

### Kernel Version

n/a

### Kubernetes Version

n/a

### Sysdump

_No response_

### Relevant log output

_No response_

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by zhanghe9702**
With CEP CRD disabled, the Cilium K8s pod watcher watches for all pods rather than just the local pods,
it's possible for Cilium to think that an endpoint that previously existed locally but moved to
another node is a validate endpoint to restore locally, this is wrong, this commit fix it.

Fixes: https://github.com/cilium/cilium/issues/18923

```release-note
cilium: Fix node mismatch endpoint restoration bug when the CiliumEndPoint CRD is disabled.
```

### CODE REVIEW

**Comment by nathanjsweet** [comment]
@zhanghe9702 please add fill out the release note section of your first comment. Thanks!

File: daemon/cmd/state.go [hunk 1]
```diff
@@ -66,9 +67,13 @@ func (d *Daemon) validateEndpoint(ep *endpoint.Endpoint) (valid bool, err error)
...
 	if ep.K8sPodName != "" && ep.K8sNamespace != "" && k8s.IsEnabled() {
 		d.k8sWatcher.WaitForCacheSync(watchers.K8sAPIGroupPodV1Core)
-		_, err = d.k8sWatcher.GetCachedPod(ep.K8sNamespace, ep.K8sPodName)
+		pod, err := d.k8sWatcher.GetCachedPod(ep.K8sNamespace, ep.K8sPodName)
 		if err != nil && k8serrors.IsNotFound(err) {
 			return false, fmt.Errorf("Kubernetes pod %s/%s does not exist", ep.K8sNamespace, ep.K8sPodName)
+		} else if err == nil && pod.Spec.NodeName != nodeTypes.GetName() {
+			// if flag `option.Config.K8sEventHandover` is false and CiliumEndpointCRD is disabled,
+			// `GetCachedPod` may return endpoint has moved to another node.
+			return false, fmt.Errorf("Kubernetes pod %s/%s does not owned by this agent", ep.K8sNamespace, ep.K8sPodName)
```
Comment (nathanjsweet) [inline daemon/cmd/state.go]:
```suggestion
			return false, fmt.Errorf("Kubernetes pod %s/%s is not owned by this agent", ep.K8sNamespace, ep.K8sPodName)
```

**Review by nathanjsweet — COMMENTED** [review]
Very good catch. I think that this will fix a number of bugs.

**Review by nathanjsweet — APPROVED** [review]

**Review by jrajahalme — APPROVED** [review]
Nit Would be nice to have a description of the change and the `Fixes` reference in the PR description.

**Review by christarazi — APPROVED** [review]

**Pull request merged**